src/org-units-selector/OrgUnitsSelector.jsx

```diff
@@ -86,6 +89,46 @@ export default class OrgUnitsSelector extends React.Component {
         });
     }
 
+    getRoots({ filter }) {
```

Comment:
[Proposal] [Untested]

```
    getRoots({ filter } = {}) {
        const { d2, listParams, rootIds } = this.props;

        const pageSize = !!rootIds ? 1000 : 10;
        const idFilter = !filter && rootIds ? `id:in:[${rootIds.join(",")}]` : null;

        const listOptions = _.omitBy(
            {
                paging: !!filter,
                pageSize: !!filter ? pageSize : null,
                level: !filter && !rootIds ? 1 : null,
                filter: filter || idFilter,
                fields: "id,displayName,path",
                ...listParams,
            },
            _.isNull
        );

        // We cannot both filter by name and check inclusion on rootIds on the same request, so
        // let's make a request filtering only by name and later check the rootIds in the response.
        const postFilter =
            filter && rootIds
                ? orgUnits =>
                      _(orgUnits)
                          .filter(orgUnit => rootIds.some(ouId => orgUnit.path.includes(ouId)))
                          .take(pageSize)
                          .value()
                : null;

        return d2.models.organisationUnits
            .list(listOptions)
            .then(collection => collection.toArray())
            .then(postFilter || _.identity);
    }
```

Reply:
Note: This is a mirror from the if/else logic in the commit, maybe I've misunderstood part of https://github.com/EyeSeeTea/vaccination-app/pull/237 and the conditions can be simplified more.

Reply:
This is a typical conditional code which can be implemented in these two ways: 1) as a true table (as I did) or 2) yours, isolated code for each property. I know the true table was some overlapping output, but my hope was that it was more clear this way.

Reply:
Okay, looks good to me then.